packages/ui/components/input-amount-dropdown/src/LionInputAmountDropdown.js

```diff
@@ -0,0 +1,460 @@
+ * @typedef {import('../../select-rich/src/LionSelectRich.js').LionSelectRich} LionSelectRich
+ * @typedef {import('../../overlays/src/OverlayController.js').OverlayController} OverlayController
+ * @typedef {TemplateDataForDropdownInputAmount & {data: {regionMetaList:RegionMeta[]}}} TemplateDataForIntlInputAmount
+ */
+
+/**
+ * LionInputTelDropdown renders a dropdown like element next to the text field, inside the
+ * prefix slot. This could be a LionSelect, a LionSelectRich or a native select.
+ * By default, the native `<select>` element is used for this, so that it's as lightweight as
+ * possible. Also, it doesn't need to be a `FormControl`, because it's purely a helper element
+ * to provide better UX: the modelValue (the text field) contains all needed info, since it's in
+ * `e164` format that contains all info (both region code and national phone number).
+ *
+ * @customElement lion-input-amount-dropdown
+ */
+export class LionInputAmountDropdown extends LionInputAmount {
+  /**
+   * @configure LitElement
+   * @type {any}
+   */
+  static properties = {
+    preferredCurrencies: { type: Array },
+    allowedCurrencies: { type: Array },
+    activeCurrency: { type: String },
+  };
+
+  refs = {
+    /** @type {DropdownRef} */
+    dropdown: /** @type {DropdownRef} */ (createRef()),
+  };
+
+  /**
+   * This method provides a TemplateData object to be fed to pure template functions, a.k.a.
+   * Pure Templates™. The goal is to totally decouple presentation from logic here, so that
+   * Subclassers can override all content without having to loose private info contained
+   * within the template function that was overridden.
+   *
+   * Subclassers would need to make sure all the contents of the TemplateData object are implemented
+   * by making sure they are coupled to the right 'ref' ([data-ref=dropdown] in this example),
+   * with the help of lit's spread operator directive.
+   * To enhance this process, the TemplateData object is completely typed. Ideally, this would be
+   * enhanced by providing linters that make sure all of their required members are implemented by
+   * a Subclasser.
+   * When a Subclasser wants to add more data, this can be done via:
+   * @example
+   * ```js
+   * get _templateDataDropdown() {
+   *   return {
+   *     ...super._templateDataDropdown,
+   *     myExtraData: { x: 1, y: 2 },
+   *   }
+   * }
+   * ```
+   * @overridable
+   * @type {TemplateDataForDropdownInputAmount}
+   */
+  get _templateDataDropdown() {
+    const localizeManager = getLocalizeManager();
+    const refs = {
+      dropdown: {
+        ref: this.refs.dropdown,
+        props: {
+          style: `height: 100%;`,
+        },
+        listeners: {
+          change: this._onDropdownValueChange,
+          'model-value-changed': this._onDropdownValueChange,
+        },
+        labels: {
+          selectCountry: localizeManager.msg('lion-input-tel:selectCountry'),
```

Comment:
This still points to the input tel :) 

I will requests translations for these 2 sentences. This will take some time, for now we should be fine with only the english text.

```suggestion
          selectCurrency: localizeManager.msg('lion-input-amount-dropdown:selectCurrency'),
```